src/interpreter.c

```diff
@@ -210,7 +210,7 @@ static jl_value_t *eval_value(jl_value_t *e, interpreter_state *s)
         jl_value_t *val = eval_value(jl_fieldref_noalloc(e, 0), s);
 #ifndef JL_NDEBUG
         JL_GC_PUSH1(&val);
-        jl_typeassert(val, jl_fieldref_noalloc(e, 1));
+        jl_typeassert(val, jl_widen_core_extended_info(jl_fieldref_noalloc(e, 1)));
```

Comment:
`analyzegc` complains on this line. Could someone tell me how I can avoid it?

Reply:
I didn't look at the error message, but I assume the issue is that `jl_widen_core_extended_info` can return a freshly-allocated result in the `Type` case, which isn't rooted anywhere. You can fix that by just changing the `JL_GC_PUSH1` to `JL_GC_PUSH2` above (on some other variable `jl_value_t *typ = NULL` (note the value at PUSH time matters, it needs to always be a valid pointer or `null` between the push and the pop)). Then just assign the result of the widening to that variable and it'll be rooted for the call.

Reply:
Thanks so much for your help! That seems to work.